src/java/org/apache/cassandra/db/commitlog/CommitLogSegment.java

```diff
@@ -175,7 +177,10 @@ static long getNextId()
 
         try
         {
-            channel = FileChannel.open(logFile.toPath(), StandardOpenOption.WRITE, StandardOpenOption.READ, StandardOpenOption.CREATE);
+            if(commitLog.configuration.isDirectIOEnabled())
+                channel = FileChannel.open(logFile.toPath(), StandardOpenOption.WRITE, StandardOpenOption.READ, StandardOpenOption.CREATE, ExtendedOpenOption.DIRECT);
```

Comment:
Does this also change the compressed and encrypted segments to use non-cached access? Could we also benchmark LZ4-compressed commit log with and without the flag?

Reply:
Direct I/O failed to run with LZ4-compressed and encrypted segments as expected due to alignment. Good to fix once this PR is upstreamed.

Reply:
Screenshot of reported error.

![image](https://github.com/apache/cassandra/assets/71302734/684a7362-5f26-4e9b-899e-f2988e9242a5)

Reply:
Okay, I see that I did not understand how Java NIO's direct mode works and we need to implement the paging/buffering ourselves, which is not as trivial.
One thing we can do for now is to only apply the direct mode if compression or encryption are not enabled, falling back to standard access otherwise (possibly with a message).